plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/async/OpenSearchThreadFactory.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.s3.async;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * OpenSearchThreadFactory is an extension of {@link ThreadFactory}
+ * that overrides new thread creation and sets them as daemon threads
+ */
+public class OpenSearchThreadFactory implements ThreadFactory {
```

Comment:
should we rename to `DeamonThreadFactory` ?

Reply:
This was copied from a package private class present within OpenSearchExecutors class. Did not want to touch that so copied it over.

Reply:
should we make that public instead ?   The Factory would be exposed to plugins as well , which should be fine IMO .

Reply:
Can we reuse `OpenSearchExecutors` instead and avoid creating something specific to plugins